lasagne/objectives.py

```diff
@@ -198,3 +202,145 @@ def aggregate(loss, weights=None, mode='mean'):
+
+    Notes
+    -----
+    This is an alternative to the binary cross-entropy loss for binary
+    classification problems
+    """
+    if targets.ndim != predictions.ndim:
+        raise TypeError('rank mismatch between targets and predictions')
+    if binary:
+        targets = 2 * targets - 1
+    return theano.tensor.nnet.relu(delta - predictions * targets)
+
+
+def mutliclass_hinge_loss(predictions, targets, delta=1):
+    """Computes the multi-class hinge loss between predictions and targets.
+
+    .. math:: L_i = \\max_{j \not = p_i} (0, t_j - t_{p_i} + \\delta)
+
+    Parameters
+    ----------
+    predictions : Theano 2D tensor
+        Predictions in (0, 1), such as softmax output of a neural network,
+        with data points in rows and class probabilities in columns.
+    targets : Theano 2D tensor or 1D tensor
+        Either a vector of int giving the correct class index per data point
+        or a 2D tensor of 1 hot encoding of the correct class in the same
+        layout as predictions
+    delta : scalar, default 1
+        The hinge loss margin
+    Returns
+    -------
+    Theano 1D tensor
+        An expression for the element-wise multi-class hinge loss
+
+    Notes
+    -----
+    This is an alternative to the categorial cross-entropy loss for multi-class
+    classification problems
+    """
+    num_cls = predictions.shape[1]
+    if targets.ndim == predictions.ndim - 1:
+        targets = theano.tensor.extra_ops.to_one_hot(targets, num_cls)
+    elif targets.ndim != predictions.ndim:
+        raise TypeError('rank mismatch between targets and predictions')
+    corrects = predictions[targets.nonzero()]
+    rest = theano.tensor.reshape(predictions[(1-targets).nonzero()],
+                                 (-1, num_cls-1))
+    rest = theano.tensor.max(rest, axis=1)
+    return theano.tensor.nnet.relu(rest - corrects + delta)
+
+
+def binary_accuracy(predictions, targets, threshold=0.5):
+    """Computes the binary accuracy between predictions and targets.
+
+    .. math:: L_i = \\mathbb{I}(t_i == p_i)
+
+    Parameters
+    ----------
+    predictions : Theano 1D tensor
+        Predictions in [0, 1], such as a sigmoidal output of a neural network,
+        with data points in rows and a single column representing the
+        probability of the point belonging to the positive class
+    targets : Theano 1D tensor
+        Targets in {0, 1}, such as ground truth labels.
+    threshold : A scalar, default: 0.5
+        Specifies at what threshold to consider the predictions being of the
+        positive class
+    Returns
+    -------
+    Theano 1D tensor
+        An expression for calculating the element-wise binary accuracy for each
+        data point
+
+    Notes
+    -----
+    This is a strictly non differential function as it includes an argmax.
+    This objective function should never be used with a gradient calculation.
+    It is intended as a convenience for validation and testing not training.
```

Comment:
Almost, there's no argmax in here :) For `categorical_accuracy`, the text is fine, but for `binary_accuracy`, what about:
```
This objective function should not be used with a gradient calculation;
its gradient is zero everywhere. It is intended as a convenience for
validation and testing, not training.
```